2주차 김종현

sunday00 opened the pull request:
# 2주차 미션 제출 (이름)

## 수행인증영상 유튜브 링크
https://www.youtube.com/watch?v=yUqA_cBJPrs

* 유튜브 링크 : 주소를 적어주세요.
  * 추가과제 1번(도커)를 수행했다면, 이 영상에 꼭 그 증거를 담아주세요.
    * 증거는 도커를 실행해보는 것으로 충분합니다.

## 과제 수행 여부를 적어주세요.
## 필수과제
- [x] 게시물 좋아요기능을 구현해주세요.
  - 특이사항 : 없음

## 추가과제
- [x] VirtualBox로 리눅스를 설치하고 그 안에 도커를 설치해주세요.
  - 특이사항 : 없음
- [x] 리액트로 할일리스트를 구현해주세요.
  - 특이사항 : 코드펜 내키지 않아서 소스에 포함되어 있습니다. 프로젝트루트/todolist 안에 소스가 있습니다.

sunday00 committed:
start

sunday00 committed:
prepare main

sunday00 committed:
write my working note on readme.md

sunday00 committed:
make migration

sunday00 committed:
store article with 0~3 tags

sunday00 committed:
some style and separate component

sunday00 committed:
list and read service and controller complete

sunday00 committed:
list and read complete

sunday00 committed:
update complete

sunday00 committed:
crud complete

sunday00 committed:
meilisearch added

sunday00 committed:
create bug fix

sunday00 committed:
message for user

sunday00 committed:
search with tag available

sunday00 committed:
git ignore data.ms

sunday00 committed:
complete

sunday00 committed:
storage

sunday00 committed:
:tada: clone from week1

clone from week1

sunday00 committed:
:lipstick: some ui and layout should into breeze

some ui and layout should into bre

sunday00 committed:
:white_check_mark: auth feature test should pass by change redirect

auth feature test should pass by c

sunday00 committed:
:construction: not logged user process to redirect or abort

not logged user process to redirec

sunday00 committed:
:white_check_mark: update articlefactory adding user

update articlefactory adding user

sunday00 committed:
:white_check_mark: article tests pass needing auth

article tests pass needing auth

sunday00 committed:
:white_check_mark: a little modify articlecontroller

a little modify articlecontroller

sunday00 committed:
:necktie: policy using service and some view

policy using service and some view

sunday00 committed:
:building_construction: validate in request and ...

validate in request and test config modify to disable meilisearch

sunday00 committed:
:white_check_mark: auth is needed

auth is needed tests

sunday00 committed:
:white_check_mark: more clear test code

more clear test code

sunday00 committed:
:white_check_mark: love test made

love test made

sunday00 committed:
:lipstick: love view

love view

sunday00 committed:
:sparkles: like is complete

like is complete

sunday00 committed:
:tada: frontend react

frontend react

sunday00 committed:
:necktie: frontend done

frontend react done

aihoshistar commented:
## 인상깊은 점
 - loves 테이블 네이밍 고민하는 과정에 한수 배웠습니다. 
 - 키보드에서 적축사운드가 나는 것 같습니다. 
 - 컨테이너와 서비스와 react 를 리뷰하시는 장면을 통헤 코드 한줄한줄아 깊은 생각이 들어가 계신 것 같습니다.
 - 굉장히 부지런 하시군요 4일전에 끝내시다니 

## 아쉬운점 
도커를 설치하면 `docker` 그룹이 생깁니다. 
사용자 관리를 통해 1,2 차 그룹을 추가하면 sudo를 입력하지 않으셔도 됩니다. 
```bash 
sudo usermod -aG docker $USER
newgrp docker
```

sunday00 commented:
- 과찬이십니다.
- 키보는 싸구렵입니다. 
- 역시 과찬이십니다.
- 실력이 없으면 성실하기라도 해야... ㅠㅠ

- 이 점은 저도 고민을 좀 해봤는데... 한타라도 편하게 sudo를 안쳐도 되게 하는게 좋은게 맞는 건지 잘 모르겠습니다. 어쩐지 리눅스는 아직 공부할 것도 많은데, sudo를 안치게 만드는게 정말 안전하다고 확신할 수 있는가, 4타 더 치는게 엄청 어려울까에 대한 고민은 좀 있습니다... ^^

sunday00 commented:
> ## 인상깊은 점
> * loves 테이블 네이밍 고민하는 과정에 한수 배웠습니다.
> * 키보드에서 적축사운드가 나는 것 같습니다.
> * 컨테이너와 서비스와 react 를 리뷰하시는 장면을 통헤 코드 한줄한줄아 깊은 생각이 들어가 계신 것 같습니다.
> * 굉장히 부지런 하시군요 4일전에 끝내시다니
> 
> ## 아쉬운점
> 도커를 설치하면 `docker` 그룹이 생깁니다. 사용자 관리를 통해 1,2 차 그룹을 추가하면 sudo를 입력하지 않으셔도 됩니다.
> 
> ```shell
> sudo usermod -aG docker $USER
> newgrp docker
> ```
과찬이십니다.

키보는 싸구렵입니다.

역시 과찬이십니다.

실력이 없으면 성실하기라도 해야... ㅠㅠ

이 점은 저도 고민을 좀 해봤는데... 한타라도 편하게 sudo를 안쳐도 되게 하는게 좋은게 맞는 건지 잘 모르겠습니다. 어쩐지 리눅스는 아직 공부할 것도 많은데, sudo를 안치게 만드는게 정말 안전하다고 확신할 수 있는가, 4타 더 치는게 엄청 어려울까에 대한 고민은 좀 있습니다... ^^

wngur6076 commented:
- 좋은점
  - laravel Breeze에 대해서 찾아보니 좋은 것 같아요.
  - LoveUpdateTest,  LoveReadTest 테스트코드를 작성하신 것을 보고 많이 배웠습니다.
  - 영상에서 리액트 컴포넌트에 설명하실 때 인상 깊었어요.

- 의견
  - 서비스에 관한 테스트코드들도 보고 싶은데 조금 아쉬워요.

sunday00 commented:
> * 좋은점
>   
>   * laravel Breeze에 대해서 찾아보니 좋은 것 같아요.
>   * LoveUpdateTest,  LoveReadTest 테스트코드를 작성하신 것을 보고 많이 배웠습니다.
>   * 영상에서 리액트 컴포넌트에 설명하실 때 인상 깊었어요.
> * 의견
>   
>   * 서비스에 관한 테스트코드들도 보고 싶은데 조금 아쉬워요.

- 라라벨은 바퀴를 새로 발명할 필요 없게 온갖 바퀴를 준비해 두었습니다. 무한궤도, 자전거바퀴, 자동차바퀴 ㅋㅋㅋ
- 그러나 사용엔 주의를 요합니다. JWT, barer, oauth, cookie, session, ... 인증 인가엔 유행이 계속 바뀌고 프론트 엔드도 부트스트랩+jQuery+반응형이 주류였다가, 어느새 바닐라+프론트프레임웤에서 TS 없이는 못 쓰는 시대가 되었습니다. 화면도 이제 갤럭시 폴드, z플립을 염두한 디자인도 생각해야 합니다. 상대적으로 버전 마이그레이션도 잘 안하는 백엔드에 비해서는 ㅋㅋㅋ
- 이런점에서는 쉽게 바꿀 수 있는 플러그인인가? 만약 다른 전략으로 바꿨을 때 설정할 게 너무 많지는 않은가? 다음 신입 개발자가 breeze를 전혀 모르는 채로 입사하면 인수인계는 쉬운가?에 대한 염두를 해야 합니다. 

- ㅠㅠ 유닛테스트는 너무 어려워요... 저도 아직 배우는 단계라... 이번에 할때 한번 도전은 해보겠습니다.

sunday00 committed:
:white_check_mark: love service unit test added

love service unit test added

sunday00 committed:
:tada: Now project can also API

Now project can also API

sunday00 committed:
:construction: di with more abstract

Dependency now injected by AbstractClass implemented interface and service with repository

sunday00 committed:
:sparkles: api love

api love

jangka2048 commented:
# 2주차 과제수행 리뷰
김종현님 안녕하세요, 장희성 강사입니다.
PR과 업로드한 동영상을 확인했습니다. 

## 좋았던 부분
- 좋아요기능의 Table부분에서, 키워드 이름과 같은 세세한 부분까지 고려해서 구성하신 부분이 좋았습니다.
-  breeze를 이용한 간단한 인증 방식도 좋았습니다.
- 피어리뷰의 docker sudo 권한 부분에 대해서, sudo 를 더 입력하는가와 안전상의 문제를 떠나 운영하고 있는 서버의 환경에 맞추어 적절히 권한부여를 해서 사용하는것이 맞다고 생각합니다.

- React Todo List에서 동영상의 말씀과 같이,상태관리는 Redux 혹은 Recoil 등을 이용하는것이 자주 사용되는 방법입니다. 금번 김종현님이 사용하신 useState 훅 역시 간단한 Todo List 예제로 사용하기에 나쁘지 않은 선택이었다고 생각합니다.
  - 또한 TodoList 역시 깔끔히 잘 구현하신것 같습니다.

매 주 과제를 임하실 때 하나하나의 요소를 깊게 생각하고 구성하는 느낌을 받았습니다.

한 주동안 작업하시느라 정말 고생 많았습니다.

감사합니다.
